slash/src/env.cc

```diff
@@ -16,6 +17,39 @@
 namespace slash {
 
 /*
+ *  Set the resource limits of a process
+ */
+bool SetMaxFileDescriptorNum(int64_t max_file_descriptor_num) {
+  // Try to Set the number of file descriptor
+  struct  rlimit limit;
+  if (getrlimit(RLIMIT_NOFILE, &limit) != -1) {
+    if (limit.rlim_cur < (rlim_t)max_file_descriptor_num) {
+      // rlim_cur could be set by any user while rlim_max are
+      // changeable only by root.
+      rlim_t previous_limit = limit.rlim_cur;
+      limit.rlim_cur = max_file_descriptor_num;
+      if(limit.rlim_cur > limit.rlim_max) {
+        limit.rlim_max = max_file_descriptor_num;
+      }
+      if (setrlimit(RLIMIT_NOFILE, &limit) != -1) {
+        log_warn("your 'limit -n ' of %d is not enough for zeppelin to start, zeppelin have successfully reconfig it to %lld", previous_limit, limit.rlim_cur);
```

Comment:
I use `int` instead of `bool`. On success, the function return `0`, if set failed, will be return `-1`, and if can not get the resource limits, return `-2`.